feedback/feedback.go

```diff
@@ -0,0 +1,187 @@
+package feedback
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/trustelem/zxcvbn/match"
+	"github.com/trustelem/zxcvbn/scoring"
+)
+
+type Feedback struct {
+	Warning     string   `json:"warning"`
+	Suggestions []string `json:"suggestions"`
+}
+
+func GetFeedback(score int, sequence []*match.Match) *Feedback {
+	// starting feedback
+	if len(sequence) == 0 {
+		return defaultFeedback()
+	}
+
+	// no feedback if score is good or great.
+	if score > 2 {
+		return emptyFeedback()
+	}
+
+	// tie feedback to the longest match for longer sequences
+	longestMatch := sequence[0]
```

Comment:
I was wondering what if the slice was empty, this would be an out-of-bands error. The [original implementation](https://github.com/dropbox/zxcvbn/blob/master/src/feedback.coffee#L21) doesn't account for this.